internal/services/containerapps/helpers/container_apps.go

```diff
@@ -2455,10 +2487,11 @@ func ContainerAppStartupProbeSchemaComputed() *pluginsdk.Schema {
 func expandContainerAppStartupProbe(input ContainerAppStartupProbe) containerapps.ContainerAppProbe {
 	probeType := containerapps.TypeStartup
 	result := containerapps.ContainerAppProbe{
-		Type:             &probeType,
-		PeriodSeconds:    pointer.To(input.Interval),
-		TimeoutSeconds:   pointer.To(input.Timeout),
-		FailureThreshold: pointer.To(input.FailureThreshold),
+		Type:                &probeType,
+		InitialDelaySeconds: pointer.To(input.InitialDelay),
+		PeriodSeconds:       pointer.To(input.Interval),
+		TimeoutSeconds:      pointer.To(input.Timeout),
+		FailureThreshold:    pointer.To(input.FailureThreshold),
```

Comment:
how come no success threshold here?

Reply:
I have a container app that is failing because I can not set a success_count_threshold to it, it will assign it an empty value.
Since I have a container app that is a public container that accepts ingress, Azure does a health check of the service using startup probe.
Since I can not configure it via terraform, my container always fails until I manually configure it on the UI to have a success threshold count to be 1, which then works fine

Reply:
according to the issue you are responding too: https://github.com/hashicorp/terraform-provider-azurerm/issues/25457
startup_probe could also have a success threshold.

Reply:
It was added initialy, but removed later in the same PR due to review in https://github.com/hashicorp/terraform-provider-azurerm/pull/27551/commits/c8002c021c2025e6f18dc03bbcfc12b75f419bf4

It was removed due to the fact that only a single value is accepted and that is `1`. You cannot set the value to anything else. I also use this code for our infra and it's working as expected. 

The question would be how it worked until now as the option wasn't present there? If it's really needed, I guess you could create a PR  with the changes from that commit to add the success threshold property and see if it's merged.

![image](https://github.com/user-attachments/assets/466daf89-31f3-4013-b442-1d92d7f64e3f)

Reply:
just tested it again and I do not get the same result - somehow success count threshold is not defaulted to 1.
<img width="571" alt="image" src="https://github.com/user-attachments/assets/265b767f-959f-466e-a786-94cb0a1012c5">

Reply:
You could create a new PR with changes from https://github.com/hashicorp/terraform-provider-azurerm/commit/c8002c021c2025e6f18dc03bbcfc12b75f419bf4 and put the screenshot as an explanation. Are you sure it fails due to the value being empty?

Reply:
Yes I am sure - because once I edit the container config on the UI, once I add a success threshold, then the container lives, otherwise it just dies and keeps restarting

Reply:
I attempted to make a PR however I get denied
```remote: Permission to hashicorp/terraform-provider-azurerm.git denied to Dmitriyx```

Reply:
You need to fork the repo and make changes in the forked repository. Afterwards you can create a PR in Github.